src/main/cpp/win_fsnotifier.cpp

```diff
@@ -148,9 +151,9 @@ Java_net_rubygrapefruit_platform_internal_jni_WindowsFileEventFunctions_startWat
 
     printf("\n~~~~ Configuring...\n");
 
-    // TODO Should this be somewhere global?
+    JavaVM* jvm;
     int jvmStatus = env->GetJavaVM(&jvm);
```

Comment:
Interestingly, `GetJavaVM()` is supposed to return a `JavaVM` _for the current thread,_ yet we can happily reuse it in the watcher thread, too. There also doesn't seem to be another way to get hold of a `JavaVM` instance, save for starting a new JVM. Maybe the documentation is incorrect?

> Returns the Java VM interface (used in the Invocation API) associated with the current thread. The result is placed at the location pointed to by the second argument, vm.

https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#wp23168

Reply:
Ah, it doesn't actually say that the `JavaVM` needs to be associated with only one thread. OK, I guess we use it the right way then.